cv_next/app/feed/components/filterPanel.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from "react";
+import { SearchBox } from "./filters/searchbar";
+import { DropdownInput } from "./filters/valueSelect";
+import Categories from "@/types/models/categories";
+import { useMemo } from "react";
+
+export interface filterValues {
+  searchValue: string;
+  categoryId: number | null;
+}
+
+export const FilterPanel = ({
+  defaultFilters,
+  onChange,
+  // eslint-disable-next-line unused-imports/no-unused-vars
+  cvs,
+}: {
+  defaultFilters: filterValues;
+  cvs: CvModel[];
+  onChange: (filters: filterValues) => void;
+}) => {
+  const [searchValue, setSearchValue] = useState(defaultFilters.searchValue);
+  const [categoryId, setCategoryId] = useState(defaultFilters.categoryId);
+
+  useEffect(() => {
+    onChange({
+      categoryId: categoryId,
+      searchValue: searchValue,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
There's no actual reason to ignore the exhaustive deps warning here. It's not really important, but ignoring exhaustive deps should be only in extreme cases, not a common practice. You can easily ignore something rather important you don't really know of its importance if you're used to ignoring.

Reply:
I ignored it due to the spam of reviewbot over it(as it is important) but in this case the `onchage` function that it talks about doesn't change

Reply:
The reviewbot's messages are not "spam", they're here to help us :)
I know that *currently* there's absolutely no actual need to put the on change function in the deps array, but because it's good practice to put everything *subject to change* in the deps array (because you never know what the code would look like in the future), we do want to follow it. You can look it up, but generally the consensus is always listen to the exhaustive-deps warn.

Reply:
I see. I tried putting that the onchange there but it kept reloading everything, that’s why I commented that, but I’ll make sure I delete it. Any idea about the reload thing?

Reply:
Ah, great, it means you probably need to wrap it using useCallback because it does, in fact, changes on every render :)

Reply:
Yeah lol
A bit after writing that I check some stuff and saw that’s the way to go